distributed/shuffle/_worker_extension.py

```diff
@@ -234,7 +234,7 @@ async def add_partition(
 
     @abc.abstractmethod
     async def get_output_partition(
-        self, i: T_partition_id, key: str
+        self, i: T_partition_id, key: str, **kwargs: Any
```

Comment:
Why allow a generic `**kwargs` instead of `meta=`? The code here looks functionally fine, but I have a slight preference for avoiding `**kwargs` if possible

Reply:
Could still use `meta=None` instead of `**kwargs` though, no? (even if array re-chunking doesn't choose to use it for now)

Reply:
If people like that better, we can change it. I don't have strong opinions and plan to refactor this soon.

Reply:
>I don't have strong opinions and plan to refactor this soon.

Makes sense - I don't have strong feelings either. Mostly just trying to digest the general code.  That said, I am pretty interested to know about the refactoring plans.

FYI, we are currently experimenting with a decomposed version of `P2PShuffleLayer` (i.e. `ShuffleTransfer(Blockwise) -> ShuffleBarrier(Layer) -> ShuffleUnpack(Blockwise)`).  I'd like to get feedback on the approach once https://github.com/dask/dask/pull/10312 and a modified version of https://github.com/dask/distributed/pull/7743 get in.

Reply:
I've been meaning to check those PRs out, I'll block some time for it next week.

Reply:
We're now using `meta` instead of `**kwargs`.